noir_stdlib/src/hash/sha256.nr

```diff
@@ -169,29 +164,74 @@ unconstrained fn build_msg_block<let N: u32>(
     } else {
         BLOCK_SIZE
     };
-    for k in 0..block_input {
-        msg_block[k] = msg[msg_start + k];
+
+    // Figure out the number of items in the int array that we have to pack.
+    // e.g. if the input is [0,1,2,3, 4,5] then we need to pack it as 2 items: [0123, 4500]
+    let mut int_input = block_input / INT_SIZE;
+    if block_input % INT_SIZE != 0 {
+        int_input = int_input + 1;
+    };
+
+    for i in 0..int_input {
+        let mut msg_field: u32 = 0;
```

Comment:
I switched to `u32` because I used division somewhere and with `Field` I couldn't get back the values. Does it make a difference?

Reply:
Yes, I kept it for historical reasons, but I see how a newcomer would be scratching their heads. That's why I added an `INT_FIELD` to explain and indicate in functions that also bear the "field" name what we're returning. Should we call it "item"?